core-plugin/src/main/groovy/com/github/jrubygradle/api/core/JRubyCorePlugin.groovy

```diff
@@ -0,0 +1,34 @@
+package com.github.jrubygradle.api.core
+
+import com.github.jrubygradle.api.gems.GemGroups
+import com.github.jrubygradle.internal.gems.GemVersionResolver
+import groovy.transform.CompileStatic
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.plugins.ExtensionAware
+
+/** Provides only a repository handler extensiosn for looking up rubygem
+ * metadata.
+ *
+ * @author Schalk W. Cronjé
+ *
+ * @since 2.0
+ */
+@CompileStatic
+class JRubyCorePlugin implements Plugin<Project> {
```

Comment:
The core plugin will add `gemGroups` as a project extension and `ruby` as a repositories extension. The change uses APIs thgat are friendly towards the Kotlin DSL.